### PR TITLE
[Api][Checkout] Order, shipping, promotion totals and order items on checkout summary

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | part of #11250 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Shop/CartContext.php [hunk 1]
```diff
@@ -261,6 +262,8 @@ private function putProductToCart(ProductInterface $product, string $tokenValue,
             'quantity' => $quantity,
         ]);
 
+        $this->sharedStorage->set('cart_token', $tokenValue);
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/CartContext.php:265]: Why is it needed? It seems that tokenValue is already set in sharedStorage

File: features/checkout/seeing_order_summary/seeing_items_on_order_summary_page.feature [hunk 2]
```diff
@@ -13,13 +13,13 @@ Feature: Seeing order items on order summary page
...
-    @ui
+    @ui @api
     Scenario: Seeing order items and theirs quantity on the order summary page
-        Given I have 4 products "Lannister Coat" in the cart
-        And I have 2 products "Targaryen Jacket" in the cart
-        And I have 3 products "Stark T-Shirt" in the cart
+        When I add 4 products "Lannister Coat" to the cart
+        And I add 2 products "Targaryen Jacket" to the cart
+        And I add 3 products "Stark T-Shirt" to the cart
         And I am at the checkout addressing step
```
Comment (GSadee) [inline features/checkout/seeing_order_summary/seeing_items_on_order_summary_page.feature:21]: It doesn't look like `When` step

File: features/checkout/seeing_order_summary/seeing_order_locale.feature [hunk 3]
```diff
@@ -12,10 +12,10 @@ Feature: Seeing order locale on order summary page
         And the store allows paying offline
         And I am a logged in customer
 
-    @ui
+    @ui @api
     Scenario: Seeing order locale on the order summary page
-        Given I have product "Stark T-Shirt" in the cart
-        When I proceed through checkout process
+        When I added "Stark T-Shirt" product to the cart
```
Comment (GSadee) [inline features/checkout/seeing_order_summary/seeing_order_locale.feature:17]:
```suggestion
        When I add "Stark T-Shirt" product to the cart
```

File: features/checkout/seeing_order_summary/seeing_shipping_total_on_order_summary_page.feature [hunk 4]
```diff
@@ -13,28 +13,28 @@ Feature: Seeing order shipping total on order summary page
         And the store allows paying offline
         And I am a logged in customer
 
-    @ui
+    @ui @api
     Scenario: Seeing the shipping total on order summary
-        Given I have "Guards! Guards! - book" variant of product "Guards! Guards!" in the cart
-        When I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I add "Guards! Guards! - book" variant of this product to the cart
+        And I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
```
Comment (GSadee) [inline features/checkout/seeing_order_summary/seeing_shipping_total_on_order_summary_page.feature:19]:
```suggestion
        And I specify the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
```

File: features/checkout/seeing_order_summary/seeing_summary_order_with_free_product.feature [hunk 5]
```diff
@@ -17,10 +17,10 @@ Feature: Seeing a summary of the order with free product
         And it gives "$5.00" discount to every order
         And I am a logged in customer
 
-    @ui
+    @ui @api
     Scenario: Seeing free order
-        Given I have product "Greyjoy Coat" in the cart
-        When I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I added "Greyjoy Coat" product to the cart
```
Comment (GSadee) [inline features/checkout/seeing_order_summary/seeing_summary_order_with_free_product.feature:22]:
```suggestion
        When I add "Greyjoy Coat" product to the cart
```

File: features/checkout/seeing_order_summary/seeing_items_on_order_summary_page.feature [hunk 6]
```diff
@@ -13,13 +13,13 @@ Feature: Seeing order items on order summary page
...
-        Given I have 4 products "Lannister Coat" in the cart
-        And I have 2 products "Targaryen Jacket" in the cart
-        And I have 3 products "Stark T-Shirt" in the cart
-        And I am at the checkout addressing step
-        When I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I add 4 products "Lannister Coat" to the cart
+        And I add 2 products "Targaryen Jacket" to the cart
+        And I add 3 products "Stark T-Shirt" to the cart
+        And I go to the checkout addressing step
+        And I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
```
Comment (GSadee) [inline features/checkout/seeing_order_summary/seeing_items_on_order_summary_page.feature:22]:
```suggestion
        And I specify the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
```

**Comment by GSadee** [comment]
Thank you, Adam! :1st_place_medal:

**Pull request merged**